tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -208,8 +210,47 @@ export class WebGPUBackend extends KernelBackend {
     this.maybeReleaseBuffer(dataId);
   }
 
+  private getCommandEncoder(): GPUCommandEncoder {
+    // If we're in the middle of a pass, end it so we can use
+    // the command encoder.
+    if (this.passEncoder !== undefined) {
```

Comment:
Could you move this logic for ending the passEncoder into a separate method and explicitly call it from the place you call getCommandEncoder, and within submitQueue? Just want getCommandEncoder to do one thing.